### PR TITLE
New: Support JSON parsing in Cardigann

### LINKED ISSUE #550: Jackett Pull - Prowlarr Cardigann V3 - Add API Support to Cardigann & Other
~~https://github.com/Jackett/Jackett/pull/12424~~
- [x] - [Core: Add Basic JSON Parsing from API ](https://github.com/Jackett/Jackett/commit/a64ebeeca897d3baf54f6a8b1e7adde27cc66fbd)
- [x] - [cardigannIndexer: add some auto-optional search fields
make description an auto-optional field, same as existing poster, imdb etc.
Also add imdbid as a search field alias of the existing imdb.
Brings it in line with existing  rageid, tmdbid, tvdbid.](https://github.com/Jackett/Jackett/commit/f5802306fa1fe4edb4c7f997100a6caebe71e989)
- [x] - [core: integration with FlareSolverr 2.0.0](https://github.com/Jackett/Jackett/commit/baac47e4bd0efffe927c02a04a7d429c06bc6a5d)
- [x] - [gui+core: imdbid can have 7 or 8 digits](https://github.com/Jackett/Jackett/commit/8af73f1bc499074226d45a9f27a057f9409ce7af)
- [x] - [core: cardigann json. add http.status not OK check](https://github.com/Jackett/Jackett/commit/664dfb090b459c6ab1fff0d43edb9d65bc780737)
- [x] - [core: cardigann json: add NoResultsMessage to Search Paths Response b…
…lock | some json servers in response to a query-not-found return a message string instead of a json object.
so this is a way to avoid a json-object-no-found error and get a graceful no-results-found state.](https://github.com/Jackett/Jackett/commit/5d00fda03e02571e458ba40bd3cd5dceb3d52478)

[AB#1773](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1773)

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
NO

#### Description
Support for JSON API parsing in Cardigann

#### Todos
- [x] Baker to finish fixing indexers

Fixes #550

### CODE REVIEW

File: src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannParser.cs [hunk 1]
```diff
@@ -469,5 +341,234 @@ public IList<ReleaseInfo> ParseResponse(IndexerResponse indexerResponse)
...
+                    break;
+                case "minimumratio":
+                    release.MinimumRatio = ParseUtil.CoerceDouble(value);
+                    value = release.MinimumRatio.ToString();
+                    break;
+                case "minimumseedtime":
+                    release.MinimumSeedTime = ParseUtil.CoerceLong(value);
+                    value = release.MinimumSeedTime.ToString();
+                    break;
+                case "imdb":
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannParser.cs:492]: need to add imdbid support as well

https://github.com/Jackett/Jackett/commit/f5802306fa1fe4edb4c7f997100a6caebe71e989#diff-76185f0a79a2c5123ec2731d21a9318e3d857edda80e5b02dbbd9e812fbcc00bR1534

ref 
https://github.com/Prowlarr/Prowlarr/issues/550

Comment (Qstick) [reply src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannParser.cs:492]: Need is a strong word

Comment (bakerboy448) [reply src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannParser.cs:492]: All the v3 indexers have those changes  and any new ones added may have those as well.

not adding it means additional work on maintaining the indexers to ensure those all get reverted now and on every cherry-pick going forward and are backed out going forward of any new indexers

**Comment by bakerboy448** [comment]
Close https://requests.prowlarr.com/b/feedback/p/torrent-ilcorsaroverde when this is merged

**Pull request merged**